docs/transformations.md

```diff
@@ -0,0 +1,106 @@
+
+To summarize the available features of the two kernels:
+
+|          | Direct transformation | Inverse transformation | Evaluation |
+|----------|-----------------|--------|-------------------|
+| DFT      | *fft*           | *ifft* | x                 |
+| Spline   | *SplineBuilder* | x      | *SplineEvaluator* |
+
+So, even if the two kernels *could* share a common API, this is not currently the case but this is more due to pratical considerations than intrinsic mathematical differences between the two methods.
+
+### General transformations theory
+
+The simplest way to describe a discrete function \f$ f \f$ is to provide its value \f$ f(x_i) \f$ at every point of the mesh which supports it.
+
+Another more general method is to define a basis \f$ \phi_j \f$ in a function space and provide the set of coefficients
+\f$ c_j \f$ which generate the discrete function. The set of coefficients is thus another way to represent the discrete
+function:
+
+\f[
+f(x_i) = \sum_j c_j \phi_j(x_i)
+\f]
+
+Note: the first method is covered by the second one in the particular case of \f$ \phi \f$ being the Dirac basis functions.
+
+The aim of a computation kernel dedicated to a transformation is to determine the values of coefficients \f$ c_j \f$
+knowing \f$ f(x_i) \f$ (and for the inverse transformation, \f$ f(x_i) \f$ knowing \f$ c_j \f$). 
+
+Additionally to the *transformation* and the *inverse transformation*, there is a third operation which can be useful:
+the *evaluation*. Indeed, more than a discrete function, coefficients \f$ c_j \f$ generate an interpolating (continuous)
+function which can be evaluated anywhere between the points of the mesh.
+
+Note: performing a (direct) transformation then an evaluation is an interpolation.
+
+Bases of function spaces can have additional properties like orthogonality
+\f$ <\phi_i|\phi_j>= \int \phi_i(x) \phi_j(x)\; dx = \delta_{ij} \f$ (thus, \f$ c_j = <f|\phi_j> \f$).
+This is the case for Fourier transform, and it provides a numerical method to perform the transformation (but not the most
+efficient one). In the general case orthogonality is not verified though, thus transformation algorithms are basis-specific.
+
+### Lexical 
+
+Here is a summary of different general terms used in the context of DDC transformations:
+
+| Term     | Meaning & commentary |
+|----------|-----------------|
+| Discrete function | Function defined on a mesh (via its values at each point of via its coefficients in a particular finite basis representation). |
+| Dimension(s) of interest | Dimension(s) along which the transformation is performed |
+| Batch dimension(s) | Dimension(s) unaffected by the transformation. It leads to a set of transformations performed independantly (along dimensions of interest). Those are embarassingly parallelizable. |
+| Interpolant functions | Same as basis functions \f$ \phi_j \f$ |
```

Comment:
Are you sure about this definition?

I found https://mathworld.wolfram.com/Interpolant.html
![image](https://github.com/CExA-project/ddc/assets/25363473/5b47fb3b-6cf8-43a1-b61c-4dfea2a45ae9)

I interpret this as meaning "interpolating function" = "interpolant function". I don't think it is the same as a basis function. The basis function does not have to agree with the function being interpolated

Reply:
You I right it looks I completly invented this